swift/mixed_language_library.bzl

```
@@ -0,0 +1,616 @@
+            be found via an include path, it will result in duplicate module
+            definition errors for downstream targets unless sandboxing or remote
+            execution is used.
+        module_name: The name of the Swift module being built.
+
+            If left unspecified, the module name will be computed based on the
+            target's build label, by stripping the leading `//` and replacing
+            `/`, `:`, and other non-identifier characters with underscores.
+        private_deps: A list of targets that are implementation-only
+            dependencies of the target being built. Libraries/linker flags from
+            these dependencies will be propagated to dependent for linking, but
+            artifacts/flags required for compilation (such as .swiftmodule
+            files, C headers, and search paths) will not be propagated.
+        swift_copts: The compiler flags for the swift library.
+        swift_defines: A list of Swift defines to add to the compilation command
+            line.
+
+            Note that unlike C-family languages, Swift defines do not have
+            values; they are simply identifiers that are either defined or
+            undefined. So strings in this list should be simple identifiers,
+            not `name=value` pairs.
+
+            Each string is prepended with `-D` and added to the command line.
+            Unlike `swift_copts`, these flags are added for the target and
+            every target that depends on it, so use this attribute with caution.
+            It is preferred that you add defines directly to `swift_copts`, only
+            using this feature in the rare case that a library needs to
+            propagate a symbol up to those that depend on it.
+        swift_srcs: The sources for the swift library.
+        textual_hdrs: The list of C, C++, Objective-C, or Objective-C++ files
+            that are included as headers by source files in this rule or by
+            users of this library. Unlike `hdrs`, these will not be compiled
+            separately from the sources.
+        umbrella_header: A `File` representing an existing umbrella header that
+            should be used in the generated module map or is used in the custom
+            module map, or `None` (the default) if the umbrella header should be
+            generated based on `hdrs`. A symlink to this header is added to an
+            include path such that `#import <ModuleName/ModuleName.h>` works for
+            this and downstream targets.
+        deps: A list of targets that are dependencies of the target being built.
+        **kwargs: Additional arguments to pass to the underlying clang and swift
+            library targets.
+    """
+    features = kwargs.pop("features", [])
+    tags = kwargs.pop("tags", [])
+    visibility = kwargs.pop("visibility", None)
+
+    internal_tags = (["manual"] + tags) if "manual" not in tags else []
```

Is the intention to always forward the `tags` to the internal targets? if so, i think we want:

```suggestion
    internal_tags = (["manual"] + tags) if "manual" not in tags else tags
```